core/src/main/java/org/mule/MuleMessageDataTypeWrapper.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule;
+
+import org.mule.api.MuleMessage;
+import org.mule.api.config.MuleProperties;
+import org.mule.api.transformer.DataType;
+import org.mule.transformer.types.MimeTypes;
+
+/**
+ * Wraps the @{link DataType} associated to a {@link MuleMessage} in order to
+ * update message properties when data type is modified and vice versa.
+ *
+ * @since 3.7.0
+ */
+public class MuleMessageDataTypeWrapper<T> implements DataType<T>
```

Comment:
Can't visability be reduced, or even private in DefaultMuleMessage? Or, does it need to be public for any reason?